### DISCUSSION TITLE
Make it possible to customize runtime directories

### DISCUSSION THREAD

**Discussion by hholst80** [helix-editor/helix#12004] [General]
The `Runtime directories` seems to be a hardcoded or an at-compile-time set parameter.

I would like to be able to customize that to my liking by setting a build-time option.

(Spoiler: I will set it to `/usr/share/helix/runtime` but you can keep it as-is if that fits you better.)

```
Config file: default
Language file: default
Log file: /root/.cache/helix/helix.log
Runtime directories: /root/.config/helix/runtime;/usr/bin/runtime
Runtime directory does not exist: /root/.config/helix/runtime
Runtime directory does not exist: /usr/bin/runtime
Clipboard provider: wl-paste+wl-copy
System clipboard provider: wl-paste+wl-copy
```

**Comment by kirawi** [comment]
You can set `$HELIX_RUNTIME`

**Comment by hholst80** [comment]
that is a great workaround. but then i need to expose that in my environment. it would be preferred if I could specify the runtime directory at compile time.

**Reply by the-mikedavis** [reply]
See https://docs.helix-editor.com/master/building-from-source.html#multiple-runtime-directories, you can set `HELIX_DEFAULT_RUNTIME` at compile time instead

**Comment by hholst80** [comment]
Why is it considering `/usr/bin/runtime` as runtime? I do not see any direct executable elf files in the runtime directory what so ever? Any such binaries should be placed in `/usr/libexec/helix`, if they existed.

**Reply by kirawi** [reply]
It's party of the default strategies to look for the runtime folder. That one in particular is where Helix looks for it within the same folder it's located in.
